Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -237,6 +228,15 @@ public void propertyChange(PropertyChangeEvent evt) {
     public BlackboardArtifactNode(BlackboardArtifact artifact, String iconPath) {
         super(artifact, createLookup(artifact, false));
         this.artifact = artifact;
+        
+        BlackboardArtifact.Type artType = null;
```

Comment:
This temp var seems unnecessary, just set this.artifactType directly.

Reply:
I made `BlackboardArtifactNode.artifactType` a final variable since it shouldn't change for the lifetime of the object similar to `BlackboardArtifactNode.artifact`.  I think assigning directly within the try/catch won't compile because it could potentially be unassigned if there was an exception.  Is it okay to leave as is given that circumstance or did you have something else in mind?

Reply:
Make a method, that takes a blackboard artifact and returns the type.  The BlackboardArtifactNode constructor already has a lot going on, moving this bit to its own method will be cleaner then all this, plus you can assign the class var without the need of a temp var.

Reply:
Ok.  Good to go.